### ISSUE TITLE
Ipython-book layer caused error

### ISSUE THREAD

**Issue by lujianmei** [syl20bnr/spacemacs#10321]
When i add ipython-book in my layer config, then i first open any of my org file, it cause an error:

`File mode specification error: (wrong-type-argument stringp sh)
`
And also all my org config were not loaded, even code in (with-eval-after-load 'org ). 
And i try to remove this layer, all things are ok.

Following is my system information

#### System Info :computer:
- OS: darwin
- Emacs: 25.3.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. c130014f)
- Graphic display: t
- Distribution: spacemacs
- Editing style: hybrid
- Completion: helm
- Layers:
```elisp
((helm :variables helm-M-x-fuzzy-match t helm-buffers-fuzzy-matching t helm-recentf-fuzzy-match t helm-move-to-line-cycle-in-source nil helm-ff-search-library-in-sexp t helm-ff-file-name-history-use-recentf t)
 auto-completion
 (spacemacs-layouts :variables layouts-enable-autosave nil layouts-autosave-delay 300)
 (better-defaults :variables better-defaults-move-to-end-of-code-first t better-defaults-move-to-beginning-of-code-first t)
 (ibuffer :variables ibuffer-group-buffers-by 'projects)
 (git :variables git-magit-status-fullscreen t magit-push-always-verify nil magit-save-repository-buffers 'dontask magit-revert-buffers 'silent magit-refs-show-commit-count 'all magit-revision-show-gravatars nil)
 markdown
 (org :variables org-enable-github-support t org-enable-bootstrap-support t)
 (osx :variables osx-use-option-as-meta t osx-dictionary-dictionary-choice "Simplified Chinese - English")
 (plantuml :variables org-ditaa-jar-path "~/dotfiles-mac/.spacemacs.d/third-plugins/ditaa0_9.jar" org-plantuml-jar-path "~/dotfiles-mac/.spacemacs.d/third-plugins/plantuml.1.2017.19.jar" plantuml-jar-path "~/dotfiles-mac/.spacemacs.d/third-plugins/plantuml.1.2017.19.jar")
 (mu4e :variables mu4e-installation-path "~/.spacemacs.d/third-plugins/mu4e" mu4e-enable-notifications t)
 ansible
 (imenu-list :variables imenu-list-auto-resize nil imenu-list-size 0.25)
 yaml emacs-lisp html
 (javascript :variables javascript-disable-tern-port-files nil)
 lua
 (python :variables python-test-runner 'pytest python-enable-yapf-format-on-save t python-auto-set-local-pyenv-version 'on-visit)
 ipython-notebook common-lisp
 (shell :variables shell-default-height 30 shell-default-term-shell "/usr/local/bin/zsh" shell-default-position 'bottom)
 docker lujianmei)
```
- System configuration features: JPEG RSVG IMAGEMAGICK NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES

Following is debug information when this error show up.

`Debugger entered--Lisp error: (wrong-type-argument stringp sh)
  intern(sh)
  ob-ipython--configure-kernel(("bash" . "bash"))
  mapcar(ob-ipython--configure-kernel (("ir" . "R") ("bash" . "bash") ("scala" . "scala") ("python3" . "python")))
  -map(ob-ipython--configure-kernel (("ir" . "R") ("bash" . "bash") ("scala" . "scala") ("python3" . "python")))
  ob-ipython-auto-configure-kernels()
  run-hooks(change-major-mode-after-body-hook text-mode-hook outline-mode-hook org-mode-hook)
  apply(run-hooks (change-major-mode-after-body-hook text-mode-hook outline-mode-hook org-mode-hook))
  run-mode-hooks(org-mode-hook)
  org-mode()
  #[514 "\303\212\214~\210eb\210\304\305\306\303#\203\254\307 \310!\311=\203\250\312\313\"\312\314\"\315\232\203X\316\317\"\203\247\320\321\"\320\322\"\206;\323\324\"\211\203K\211\241\210\202SBB\262\266\202\247\325\326\211\327!#\330!\211\203j\202m\331!\211\235\204\245\332\333!r\211q\210\334\335\336\337\340!\341\"\322$\216\204\217\342!\343 \210\344\345\"c\210\300\242	B\"\262*\210\266\210\210\202	+\207" [(#0) case-fold-search default-directory t re-search-forward "^[ 	]*#\\+\\(MACRO\\|SETUPFILE\\):" nil org-element-at-point org-element-type keyword org-element-property :value :key "MACRO" string-match "^\\(.*?\\)\\(?:\\s-+\\(.*\\)\\)?\\s-*$" match-string 1 2 "" assoc org-unbracket-string "\"" org-trim org-file-url-p expand-file-name generate-new-buffer " *temp*" make-byte-code 0 "\301\300!\205	\302\300!\207" vconcat vector [buffer-name kill-buffer] file-name-directory org-mode org-file-contents noerror] 14 "\n\n(fn FILES TEMPLATES)"](nil nil)
  org-macro--collect-macros()
  org-macro-initialize-templates()
  org-mode()
  set-auto-mode-0(org-mode nil)
  set-auto-mode()
  normal-mode(t)
  after-find-file(nil t)
  find-file-noselect-1(#<buffer sites-overall-cookie.org> "~/workspace/github/my-blog/work-notes/notes/tech-study-documents/sites-overall-cookie.org" nil nil "~/workspace/github/my-blog/work-notes/notes/tech-study-documents/sites-overall-cookie.org" (8595816566 16777220))
  find-file-noselect("/Users/kevin/workspace/github/my-blog/work-notes/notes/tech-study-documents/sites-overall-cookie.org")
  mapcar(find-file-noselect ("/Users/kevin/workspace/github/my-blog/work-notes/notes/tech-study-documents/sites-overall-cookie.org"))
  (let* ((files (helm-marked-candidates)) (buffers (mapcar (quote find-file-noselect) files))) (spacemacs//helm-open-buffers-in-windows buffers))
  (lambda (candidate) (let* ((files (helm-marked-candidates)) (buffers (mapcar (quote find-file-noselect) files))) (spacemacs//helm-open-buffers-in-windows buffers)))("/Users/kevin/workspace/github/my-blog/work-notes/notes/tech-study-documents/sites-overall-cookie.org")
  helm-execute-selection-action-1()
  helm-execute-selection-action()
  helm-internal((helm-source-projectile-buffers-list helm-source-projectile-files-list helm-source-projectile-projects) nil "[work-notes] pattern: " nil nil "*helm projectile*" nil nil nil)
  apply(helm-internal ((helm-source-projectile-buffers-list helm-source-projectile-files-list helm-source-projectile-projects) nil "[work-notes] pattern: " nil nil "*helm projectile*" nil nil nil))
  helm((helm-source-projectile-buffers-list helm-source-projectile-files-list helm-source-projectile-projects) nil "[work-notes] pattern: " nil nil "*helm projectile*" nil nil nil)
  apply(helm ((helm-source-projectile-buffers-list helm-source-projectile-files-list helm-source-projectile-projects) nil "[work-notes] pattern: " nil nil "*helm projectile*" nil nil nil))
  helm(:sources (helm-source-projectile-buffers-list helm-source-projectile-files-list helm-source-projectile-projects) :buffer "*helm projectile*" :truncate-lines nil :prompt "[work-notes] pattern: ")
  helm-projectile()
  projectile-switch-project-by-name("~/workspace/github/my-blog/work-notes/")
  #[257 "\301\302!)\207" [projectile-completion-system helm projectile-switch-project-by-name] 3 "\n\n(fn PROJECT)"]("~/workspace/github/my-blog/work-notes/")
  helm-execute-selection-action-1()
  helm-execute-selection-action()
  helm-internal(helm-source-projectile-projects nil "[-] Switch to project: " nil nil "*helm projectile*" nil nil nil)
  apply(helm-internal (helm-source-projectile-projects nil "[-] Switch to project: " nil nil "*helm projectile*" nil nil nil))
  helm(helm-source-projectile-projects nil "[-] Switch to project: " nil nil "*helm projectile*" nil nil nil)
  apply(helm (helm-source-projectile-projects nil "[-] Switch to project: " nil nil "*helm projectile*" nil nil nil))
  helm(:sources helm-source-projectile-projects :buffer "*helm projectile*" :truncate-lines nil :prompt "[-] Switch to project: ")
  helm-projectile-switch-project(nil)
  funcall-interactively(helm-projectile-switch-project nil)
  call-interactively(helm-projectile-switch-project nil nil)
  command-execute(helm-projectile-switch-project)
`

**Comment by Miciah** [comment]
Upstream issue: https://github.com/gregsexton/ob-ipython/issues/154

**Comment by ArchCST** [comment]
Has this been fixed? I'm encounter same error too

**Comment by gnohz** [comment]
exactly the same issue: once I add the ipython-notebook layer, the following error message appears: "file mode specification error (json readable error 47)

**Comment by dalanicolai** [comment]
Just to revive... same issue here

**Comment by duianto** [comment]
The linked upstream issue has a pending PR:
fix mode query in kernel configuration (fixes [#154](https://github.com/gregsexton/ob-ipython/issues/154)) https://github.com/gregsexton/ob-ipython/pull/182

Anyone with the issue could try applying the fix locally.
